Source/CarthageKit/Xcode.swift

```diff
@@ -630,26 +630,34 @@ private func build(sdk: SDK, with buildArgs: BuildArguments, in workingDirectory
 		// Specifying destination seems to be required for building with
 		// simulator SDKs since Xcode 7.2.
 		if sdk.isSimulator {
-			let destinationLookup = Task("/usr/bin/xcrun", arguments: [ "simctl", "list", "devices" ])
+			let destinationLookup = Task("/usr/bin/xcrun", arguments: [ "simctl", "list", "devices", "--json" ])
 			return destinationLookup.launch()
 				.ignoreTaskData()
 				.map { data in
-					let string = String(data: data, encoding: .utf8)!
-					// The output as of Xcode 6.4 is structured text so we
-					// parse it using regex. The destination will be omitted
-					// altogether if parsing fails. Xcode 7.0 beta 4 added a
-					// JSON output option as `xcrun simctl list devices --json`
-					// so this can be switched once 7.0 becomes a requirement.
+					let decoder = JSONDecoder()
+					// simctl returns following JSON:
+					// {"devices": {"iOS 12.0": [<simulators...>]}]
+					let jsonObject = try! decoder.decode([String: [String: [Simulator]]].self, from: data) // swiftlint:disable:this force_try
 					let platformName = sdk.platform.rawValue
-					let regex = try! NSRegularExpression( // swiftlint:disable:this force_try
-						pattern: "-- \(platformName) [0-9.]+ --\\n.*?\\(([0-9A-Z]{8}-([0-9A-Z]{4}-){3}[0-9A-Z]{12})\\)",
-						options: []
-					)
-					let lastDeviceResult = regex.matches(in: string, range: NSRange(string.startIndex..., in: string)).last
-					return lastDeviceResult.map { result in
-						// We use the ID here instead of the name as it's guaranteed to be unique, the name isn't.
-						let deviceID = string[Range(result.range(at: 1), in: string)!]
-						return "platform=\(platformName) Simulator,id=\(deviceID)"
+					let devices = jsonObject["devices"]!
+					let allTargetSimulators = devices
+						.filter { $0.key.hasPrefix(platformName) }
+					func sortedByVersion(_ osNames: [String]) -> [String] {
+						return osNames.sorted { lhs, rhs in
+							guard let lhsVersion = lhs.split(separator: " ").last.flatMap(Float.init),
+								let rhsVersion = rhs.split(separator: " ").last.flatMap(Float.init) else {
+									return lhs < rhs
+							}
+							return lhsVersion < rhsVersion
+						}
+					}
```

Comment:
I don't see any tests for this logic. Please add a test that has multiple OS versions.